lib/jobs/secure-erase-drive.js

```diff
@@ -0,0 +1,386 @@
+    assert,
+    util,
+    _,
+    CommandUtil,
+    catalogSearch,
+    Constants
+) {
+    var logger = Logger.initialize(secureEraseJobFactory);
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function SecureEraseJob(options, context, taskId) {
+        SecureEraseJob.super_.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        assert.arrayOfObject(this.options.eraseSettings, 'eraseSettings');
+
+        this.commandUtil = new CommandUtil(this.context.target);
+        this.nodeId = this.context.target;
+
+        this.eraseSettings = {};
+        this.commands = [];
+        this.hasSentCommands = false;
+    }
+
+    util.inherits(SecureEraseJob, BaseJob);
+
+    /**
+     * @memberOf SecureEraseJob
+     * @function
+     */
+    SecureEraseJob.prototype._run = function() {
+        var self = this;
+
+        self.eraseSettings = self._validateOptions();
+
+        return Promise.resolve()
+        .then(self._collectDisks())
+        .then(function(disks){
+            return catalogSearch.getDriveIdCatalogExt(self.nodeId, disks);
+        })
+        .then(self._marshalParams)
+        .then(self._formatCommands)
+        .then(function() {
+            self._subscribeRequestCommands(function() {
+                return self.handleRequest();
+            });
+
+            self._subscribeRespondCommands(function(data) {
+                self.commandUtil.handleRemoteFailure(data.tasks)
+                .spread(function() {
+                    self._done();
+                })
+                .catch(function(err) {
+                    self._done(err);
+                });
+            });
+        });
+    };
+
+    /**
+     * @memberOf SecureEraseJob
+     * @function
+     */
+    SecureEraseJob.prototype._validateOptions = function() {
+        var self = this;
+
+        return _.transform(self.options.eraseSettings, function(result, entry) {
+
+            assert.array(entry.disks, 'disks');
+            if (entry.disks.length === 0) {
+                throw new Error('disks should not be empty');
+            }
+
+            if (entry.hasOwnProperty('operation')) {
+
+                assert.string(entry.operation, 'operation');
+
+                // Make sure "tool" is assigned when "operation" is specified
+                assert.string(entry.tool, 'tool');
+
+                // Verify the operation are supported in the specified tool
+                if (!_.includes(Constants.SecureErase[entry.tool].operation, entry.operation)) {
+                    throw new Error(entry.tool + " doesn't support operation " + entry.operation);
+                }
+            }
+
+            // Turn the drive ID number into string
+            _.forEach(entry.disks, function(disk, index) {
+                if (_.isNumber(disk)) {
+                    entry.disks[index] = disk.toString();
+                }
```

Comment:
Remove this logic since catalog search https://github.com/WangWinson/on-tasks/blob/376d0ebf9acbea0f11397b6f73ae0938cd2e1a29/lib/utils/job-utils/catalog-searcher.js#L63 can accept both string and number